plugins/utils/ccc.nix

```diff
@@ -5,59 +5,252 @@
   pkgs,
   ...
 }:
+with lib;
 helpers.neovim-plugin.mkNeovimPlugin config {
   name = "ccc";
   originalName = "ccc.nvim";
   defaultPackage = pkgs.vimPlugins.ccc-nvim;
 
   maintainers = [ helpers.maintainers.JanKremer ];
 
-  settingsOptions = {
-    default_color = helpers.defaultNullOpts.mkStr "#000000" ''
-      The default color used when a color cannot be picked. It must be HEX format.
-    '';
+  settingsOptions =
+    let
+      listOfRawLua = with helpers.nixvimTypes; nullOr (maybeRaw (listOf strLua));
+      mapToRawLua = v: if isList v then map helpers.mkRaw v else v;
+    in
+    {
+      default_color = helpers.defaultNullOpts.mkStr "#000000" ''
+        The default color used when a color cannot be picked. It must be HEX format.
+      '';
 
-    highlight_mode =
-      helpers.defaultNullOpts.mkEnum
-        [
-          "fg"
-          "bg"
-          "foreground"
-          "background"
-        ]
-        "bg"
-        ''
-          Option to highlight text foreground or background. It is used to
-          `output_line` and `highlighter`.
-        '';
-
-    lsp = helpers.defaultNullOpts.mkBool true ''
-      Whether to enable LSP support. The color information is updated in the
-      background and the result is used by `:CccPick` and highlighter.
-    '';
+      inputs = mkOption {
+        type = listOfRawLua;
+        apply = mapToRawLua;
+        default = [ ];
```

Comment:
If this option is left as the default empty list, will an empty lua table be passed to `setup`? If so, will that override the plugins default `inputs`?

We would normally have the default be `null`, which (I assume) leads to the option being omitted from the lua table.

Also, if passing a value (that doesn't explicitly include the defaults) _unsets_ said defaults, we should probably make a note of that in the docs.

Reply:
Is there a reason this can't use `helpers.defaultNullOpts.mkListOf`?

Reply:
> If this option is left as the default empty list, will an empty lua table be passed to `setup`? If so, will that override the plugins default `inputs`?
> 
> We would normally have the default be `null`, which (I assume) leads to the option being omitted from the lua table.
> 
> Also, if passing a value (that doesn't explicitly include the defaults) _unsets_ said defaults, we should probably make a note of that in the docs.

Your concern is perfectly valid. But here, it is OK:
- Empty lists/attrs are indeed ignored by `toLuaObject`, so it has the same behavior as a `null` default.
- Now, why not `null` ? Well, just to simplify the `apply` function and not having to check if the value is `null` or a list.

Reply:
With `mkListOf`, you cannot have an option where the user fills a list with lua strings that will get converted to raw lua automatically.

Reply:
Thanks for explaining 👍